### DISCUSSION TITLE
Are "annotations" warranted in presence of "description"?

### DISCUSSION THREAD

**Discussion by savchenko** [taskchampion/taskchampion#161] [Q&A]
Inspired by https://github.com/djmitche/taskchampion/issues/89.

Given that task already has properties of "name", "tag" and "description", does it make much sense to add "annotation"? Wouldn't it just confuse newcomers?

**Comment by djmitche** [comment]
Note that a task doesn't have a name - just a one-line description.  And multiple tags :)

Annotations are useful in that they can add context to a task after it is created, and they merge nicely when added in multiple places at once.

One use-case is to keep a list, e.g.,

```
task add make a grocery order
task 17 annotate carrots
task 17 annotate chicken
task 17 annotate milk
```

Another use-case is for recording updates to an ongoing task; for example:

```
task add find a fence contractor
...
task 82 annotate called jimmies fences, left message
task 82 annotate called fences r us, left message
task 82 wait:2d
...
task 82 annotate jimmies called back, will do estimate tomorrow
...
```

If we kept this information in the description, then the description would get pretty long!  It also wouldn't merge very nicely (last edit wins).
